src/confdb/data/ReleaseVersionInfo.java

```diff
@@ -0,0 +1,132 @@
+            throw new IllegalArgumentException("release tag "+releaseTag+" is not of format CMSSW_A_B_C where A,B, C are ints (although C can be X) and thus can not be parsed");
+        }
+    }
+
+    private boolean setValues(String releaseTag){
+        String releaseTagTrimmed = releaseTag;
+        if(!releaseTag.startsWith("CMSSW")){
+            int index = releaseTag.indexOf("CMSSW");
+            if(index!=-1){
+                releaseTagTrimmed = releaseTag.substring(index);
+            }
+        }
+
+        String[] splitTag = releaseTagTrimmed.split("_");
+        
+        if(splitTag.length<4) return false;
+        if(!splitTag[0].equals("CMSSW")) return false;
+
+        try {
+            cycle = Integer.parseInt(splitTag[1]);
+            major = Integer.parseInt(splitTag[2]);
+            if (splitTag[3].equals("X")){ //for integration builds
+                minor = -1;
+            }else{
+                minor = Integer.parseInt(splitTag[3]);
+            }
+        } catch (NumberFormatException ex) {            
+            return false;            
+        }
+
+        if(splitTag.length>4){            
+            if(splitTag[4].startsWith("pre")){
+                try{                    
+                    prerel = Integer.parseInt(splitTag[4].substring(3));
+                } catch(NumberFormatException ex){
+
+                }
+            }
+            if(splitTag[4].startsWith("patch")){
+                try{
+                    patch = Integer.parseInt(splitTag[4].substring(5));
+                } catch(NumberFormatException ex){
+                    
+                }
+            }
+        }
+        //System.err.print("parsed "+releaseTag);
+        //System.err.println("  "+cycle+" "+major+" "+minor+" pre "+prerel+" patch "+patch+" ");
+        return true;
+
+    }
+    public Integer cycle(){return cycle;}
+    public Integer major(){return major;}
+    public Integer minor(){return minor;} 
+    public Integer patch(){return patch;} 
+    public Integer prerel(){return prerel;} 
+    /** 
+     * checks if a given release is >= the specified values
+     * 
+     * notes:
+     *  a prerel is below its minor version
+     *  a patch is above its minor version
+     *  a nightly (X) is undefined, defaults to above as have to do something  
+     */
+    public boolean geq(int cycle,int major,int minor,int prerel,int patch){
+        if(this.cycle > cycle) return true;
+        else if(this.cycle==cycle && this.major > major) return true;
+        else if(this.cycle==cycle && this.major == major) {
+            if(minor==-1 || this.minor>minor) return true;
+            else if(this.minor==minor){
+                if(prerel!=-1){
+                    return this.prerel==null || this.prerel >= prerel;                    
+                }else if(patch!=-1){
+                    return this.patch!=null && this.patch>=patch;
+                }else{
+                    return true;
```

Comment:
```suggestion
                    return (this.patch != null || this.prerel == null);
```
I tried (by eye) to do the case `[this=X_Y_Z_preA].geq(X,Y,Z,-1,-1)`, and I got to `true` in the current implementation, while I was expecting false.

Reply:
thanks Marino! I missed that case :(   I've retested it and it now behaves as expected

```
release CMSSW_12_3_0_patch1 geq 12_3_0 true
release CMSSW_12_3_0_patch1 geq 12_3_0_patch2 false
release CMSSW_12_3_0_patch1 geq 12_3_0_pre2 true
release CMSSW_12_3_0_patch2 geq 12_3_0 true
release CMSSW_12_3_0_patch2 geq 12_3_0_patch2 true
release CMSSW_12_3_0_patch2 geq 12_3_0_pre2 true
release CMSSW_12_3_0_pre1 geq 12_3_0 false
release CMSSW_12_3_0_pre1 geq 12_3_0_patch2 false
release CMSSW_12_3_0_pre1 geq 12_3_0_pre2 false
release CMSSW_12_3_0_pre2 geq 12_3_0 false
release CMSSW_12_3_0_pre2 geq 12_3_0_patch2 false
release CMSSW_12_3_0_pre2 geq 12_3_0_pre2 true
release CMSSW_12_3_0 geq 12_3_0 true
release CMSSW_12_3_0 geq 12_3_0_patch2 false
release CMSSW_12_3_0 geq 12_3_0_pre2 true
release CMSSW_12_3_1 geq 12_3_0 true
release CMSSW_12_3_1 geq 12_3_0_patch2 true
release CMSSW_12_3_1 geq 12_3_0_pre2 true
release CMSSW_12_2_0_patch1 geq 12_3_0 false
release CMSSW_12_2_0_patch1 geq 12_3_0_patch2 false
release CMSSW_12_2_0_patch1 geq 12_3_0_pre2 false
release CMSSW_12_2_0_pre2 geq 12_3_0 false
release CMSSW_12_2_0_pre2 geq 12_3_0_patch2 false
release CMSSW_12_2_0_pre2 geq 12_3_0_pre2 false
```